File: factoryDesign/factory.py

class Employee:
    def salary(self):
        pass

class WebDeveloper(Employee):
    def salary(self):
        return 50000

class AndroidDeveloper(Employee):
    def salary(self):
        return 60000

class EmployeeFactory:
    @staticmethod
    def get_employee(empType):
        if empType.trim() == "AndroidDeveloper":
            return AndroidDeveloper()
        elif empType.trim() == "WebDeveloper":
            return WebDeveloper()
        else:
            return None

class Client:
    @staticmethod
    def main():
        emp = EmployeeFactory.get_employee("AndroidDeveloper")
        sal = emp.salary()
        print(f"Salary: {sal}")

Client.main()

File: factoryDesign/factory.cs
using System;

public interface Employee{
    int salary();
}

public class WebDeveloper : Employee
{
    public int salary(){
        return 50000;
    }
}

public class AndroidDeveloper : Employee
{
    public int salary(){
        return 60000;
    }
}

public class EmployeeFactory{
    public static Employee getEmployee(string empType){
        if (empType == "AndroidDeveloper"){
            return new AndroidDeveloper();
        }
        else if (empType == "WebDeveloper"){
            return new WebDeveloper();
        }
        else{
            return null;
        }
    }
}

public class Client{
    public static void Main(string[] args){
        Employee emp = EmployeeFactory.getEmployee(empType:"AndroidDeveloper");
        int sal = emp.salary();
        Console.WriteLine("Salary: " + sal);
    }
}
